src/vs/workbench/contrib/themes/browser/themes.contribution.ts

```diff
@@ -382,6 +382,35 @@ registerAction2(class extends Action2 {
 	}
 });
 
+const SelectColorThemeByNameCommandId = 'workbench.action.selectThemeByName';
+
+registerAction2(class extends Action2 {
+
+	constructor() {
+		super({
+			id: SelectColorThemeByNameCommandId,
+			title: localize('selectThemeByName.label', "Color Theme By Name"),
+			category: CATEGORIES.Preferences
+		});
+	}
+
+	override async run(accessor: ServicesAccessor, newTheme: any) {
+		const themeService = accessor.get(IWorkbenchThemeService);
+		themeService.getColorThemes().then(results => {
+			const filteredTheme: IWorkbenchColorTheme | undefined = results.find(
+				function (results) {
+					return results.settingsId === `${newTheme}`;
+				}
+			);
+			if (filteredTheme === undefined) {
+				console.log(localize('selectThemeByName.errorLog', "No theme found that matches ") + `\"${newTheme}\"`);
+			} else {
+				themeService.setColorTheme(filteredTheme, 'auto').then();
```

Comment:
The `then` is unnecessary.

Use `await` to wait for the result of `setColorTheme`